src/js/component.js

```diff
@@ -100,38 +102,10 @@ class Component {
     this.childIndex_ = {};
     this.childNameIndex_ = {};
 
-    let SetSham;
-
-    if (!window.Set) {
-      SetSham = class {
-        constructor() {
-          this.set_ = {};
-        }
-        has(key) {
-          return key in this.set_;
-        }
-        delete(key) {
-          const has = this.has(key);
-
-          delete this.set_[key];
-
-          return has;
-        }
-        add(key) {
-          this.set_[key] = 1;
-          return this;
-        }
-        forEach(callback, thisArg) {
-          for (const key in this.set_) {
-            callback.call(thisArg, key, key, this);
-          }
-        }
-      };
-    }
-
-    this.setTimeoutIds_ = window.Set ? new Set() : new SetSham();
-    this.setIntervalIds_ = window.Set ? new Set() : new SetSham();
-    this.rafIds_ = window.Set ? new Set() : new SetSham();
+    this.setTimeoutIds_ = new Set();
+    this.setIntervalIds_ = new Set();
+    this.rafIds_ = new Set();
+    this.namedRafs_ = new Map();
```

Comment:
At first I was wondering where the checks for the built-ins went but I see that we export the native method or our Sham from the `set.js` and `map.js` files which definitely makes it easier to use them, 👍